card/mod/carrierwave/lib/carrier_wave/file_card_uploader.rb

```diff
@@ -62,43 +165,55 @@ def extension
     end
 
     # generate identifier that gets stored in the card's db_content field
+    # @param opts [Hash] generate an identifier using the given storage options
+    #   instead of the storage options derived from the model and
+    #   the global configuration
+    # @option opts [Symbol] storage_type
+    # @option opts [String] mod
+    # @option opts [Symbol] bucket
     def db_content opts={}
-      return "" unless file.present?
-      model.load_from_mod = opts[:mod] if opts[:mod] && !model.load_from_mod
-      "%s/%s" % [file_dir, url_filename(opts)]
+      model.with_storage_options opts do
+        return model.content if model.web?
+        return "" unless file.present?
+        "%s/%s" % [file_dir, url_filename]
+      end
     end
 
     def url_filename opts={}
-      model.load_from_mod = opts[:mod] if opts[:mod] && !model.load_from_mod
-
-      basename = if (mod = mod_file?)
-                   "#{mod}#{extension}"
-                 else
-                   "#{action_id}#{extension}"
-                 end
+      model.with_storage_options opts do
+        if model.coded?
+          "#{model.mod}#{extension}"
+        else
+          "#{action_id}#{extension}"
+        end
+      end
     end
 
     def url opts={}
+      if model.cloud?
+        file.url
+      elsif model.web?
```

Comment:
as far as I know ruby doesn't support something like that.
You can do 
```
case model
when ->(m) { m.cloud? }
```
or override the === operator for symbols.